backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/record/warning/RecordWarningRepository.java

```diff
@@ -67,6 +69,23 @@ private IQueryBuilder<I_AD_Record_Warning> toSqlQuery(final RecordWarningQuery q
 			queryBuilder.addEqualsFilter(I_AD_Record_Warning.COLUMNNAME_AD_BusinessRule_ID, query.getBusinessRuleId());
 		}
 
+		if(query.getSeverity() != null)
+		{
+			queryBuilder.addEqualsFilter(I_AD_Record_Warning.COLUMNNAME_Severity, query.getSeverity().getCode());
+		}
+
 		return queryBuilder;
 	}
+
+	public boolean hasRecordWarningsWithErrorSeverity(@NonNull final TableRecordReference recordRef)
```

Comment:
i think u can simply call it hasErrors because
* u are already in RecordWarningRepository => so it's obvoius what are we talking about
* have something short but still enough descriptive